js/script.js

```diff
@@ -0,0 +1,104 @@
+var counter = 0;
+var intervalId;
+
+$(document).ready(function() {
+
+	$('#reddit-search').on('click', function(e) {
+	    e.preventDefault();
+	    search($('#input').val());
+	    console.log('clicked'+$('#input').val())
+	    $('#input').val('');
+  	})
+
+  		$('#stop-search').on('click', function(e) {
+			    e.preventDefault();
+			    $('#images').html('');
+
+			    clearInterval(intervalId);
+
+  			})
+
+	function search(input) {
+		$.get('http://www.reddit.com/search.json', {q:input},"+nsfw:no").done(function(data) {
+	   		console.log(data);
+
+	   		var children = data.data.children;
+	   		console.log(children);
+
+	   		var hasUrl = function(child){
```

Comment:
This still gets you posts without pictures. You have to check your data a bit more to be sure your assumptions are correct.